framework/wazuh/core/cluster/common.py

```diff
@@ -955,7 +960,32 @@ def receive_file(self, data: bytes) -> Tuple[bytes, bytes]:
         bytes
             Response message.
         """
-        self.in_file[data] = {'fd': open(common.WAZUH_PATH + data.decode(), 'wb'), 'checksum': hashlib.sha256()}
+        # Decode path requested by peer node
+        rel = data.decode()
+
+        # Paths must be absolute within WAZUH_PATH
+        if not rel.startswith("/"):
+            return b"err", b"Invalid path format"
+
+        # Normalize path and block traversal
+        rel_norm = posixpath.normpath(rel)
+        if ".." in PurePosixPath(rel_norm).parts:
+            return b"err", b"Path traversal detected"
+
+        # Allowlist: only permit cluster data directories
+        if not any(rel_norm.startswith(p) for p in _ALLOWED_PREFIXES):
+            return b"err", b"Write path not allowed"
+
+        # Resolve final destination and ensure it stays inside WAZUH_PATH
+        dst = os.path.realpath(os.path.join(common.WAZUH_PATH, rel_norm.lstrip("/")))
```

Comment:
How about directly joining the path with os.path. join and then check if realpath is any of the common.WAZUH_PATH + _ALLOWED_PREFIXES combinations